share/namespace.go

```diff
@@ -2,13 +2,35 @@ package share
 
 import (
 	"bytes"
+	"encoding/hex"
+	"encoding/json"
 	"fmt"
 )
 
 type Namespace struct {
 	data []byte
 }
 
+// MarshalJSON encodes namespace to the json encoded bytes.
+func (n Namespace) MarshalJSON() ([]byte, error) {
+	return json.Marshal(n.data)
```

Comment:
We should consider making json marshalling human readable, like in hex instead.

Reply:
Although, if that breaks something, then its not really worth it rn

Reply:
Why should marshaling be human readable? This is a byte representation that will be sent over the wire. I've added `String()` method to the namespace, which makes it human-readable.

Reply:
I mean it's nice to keep human readable json human readable, but that's more of a general comment for everyone and a very minor concern, so resolving